src/20_Components/amp-dynamic-css-classes.html

```diff
@@ -0,0 +1,58 @@
+  -->
+  <script async custom-element="amp-dynamic-css-classes" src="https://cdn.ampproject.org/v0/amp-dynamic-css-classes-0.1.js"></script>
+  <script async custom-element="amp-social-share" src="https://cdn.ampproject.org/v0/amp-social-share-0.1.js"></script>
+  <link rel="canonical" href="<%host%>/components/amp-dynamic-css-classes/" >
+  <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+  <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
+  <script async src="https://cdn.ampproject.org/v0.js"></script>
+  <style amp-custom>
+    body:not(.amp-referrer-pinterest-com) .if-pinterest, 
+    body:not(.amp-referrer-ampbyexample-com) .if-ampbyexample, 
+    body:not(.amp-referrer-google-com) .if-google,
+    body:not(.amp-referrer-twitter-com) .if-twitter,
+    body:not(.amp-referrer-localhost) .if-localhost {
+      display: none;
+    }
+    body:not(.amp-viewer) .if-viewer,
+    body.amp-viewer .if-not-viewer {
+      display: none;
+    }
+  </style>
+</head>
+<body>
+
+  <!-- You do not need to include a call to the amp-dynamic-css-classes component to initialize the classes on the body element -->
+
+  <!-- #### Referrer -->
+  <!--
+    Using these referrer classes can allow you to conditionally include elements.
+  -->
+  <span class="if-pinterest">You were referred here or embedded by Pinterest!  I'll add a few extra "Pin It" buttons</span>
+  <span class="if-twitter">You were referred here or embedded by Twitter!  I'll add a few extra Tweet buttons</span>
+  <span class="if-google">You were referred here or embedded by Google!  I'll add a few extra Google+ buttons</span>
+  <span class="if-ampbyexample">You came here directly!  Cool :)</span>
+  <span class="if-localhost">You came here directly!  Cool :)</span>
+
+  <amp-social-share type="email" layout="responsive" height="30" width="30"></amp-social-share>
+  <amp-social-share type="twitter" layout="responsive" height="30" width="30" class="if-twitter"></amp-social-share>
+  <amp-social-share type="pinterest" layout="responsive" height="30" width="30" class="if-pinterest"></amp-social-share>
+  <amp-social-share type="linkedin" layout="responsive" height="30" width="30" class="if-linkedin"></amp-social-share>
+
+  <!-- #### Viewer -->
```

Comment:
Well, the Google AMP Viewer currently redirects to the canonical if viewed from a Desktop UA, so I'm not sure it'd be a meaningful exercise for desktop viewers.  

One thing we could do is provide the link and suggest users use mobile emulation mode in Chrome.  Would that work?